pkg/cmd/roachtest/operations/add_database.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package operations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+type cleanupAddedDatabase struct {
+	db string
+}
+
+func (cl *cleanupAddedDatabase) Cleanup(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) {
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+
+	o.Status(fmt.Sprintf("dropping database %s", cl.db))
+	_, err := conn.ExecContext(ctx, fmt.Sprintf("DROP DATABASE %s", cl.db))
+	if err != nil {
+		o.Fatal(err)
+	}
+}
+
+func runAddDatabase(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) registry.OperationCleanup {
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+
+	rng, _ := randutil.NewPseudoRand()
+	withOpts := ""
+	if rng.Float64() < 0.25 {
+		withOpts += " ENCODING = 'UTF-8'"
+	}
+	if rng.Float64() < 0.25 {
+		withOpts += " CONNECTION LIMIT = -1"
+	}
+	if rng.Float64() < 0.5 {
+		roleName := pickRandomRole(ctx, o, conn)
+		withOpts += fmt.Sprintf(" OWNER = %s", roleName)
```

Comment:
More of a question for me: does this mean we'll not be able to drop the db unless we're that OWNER?

Reply:
That's a great question. I think that's correct, but the database team would have more accurate information. From what I understand, if the OWNER is not specified, the current user automatically becomes the owner of the database. The fact that a user with root privileges can drop databases they don't own suggests that there are certain administrative privileges in play.

Reply:
Yeah I was more worried about us assigning a random other role as an owner, and then not being able to drop it ourselves. Maybe the caller running with admin privileges is a safe assumption elsewhere too and this is fine. Feel free to merge away